clients/spark/core/src/main/scala/io/treeverse/clients/BulkRemoverFactory.scala

```diff
@@ -65,20 +68,37 @@ object BulkRemoverFactory {
 
   private class S3BulkRemover(hc: Configuration, storageNamespace: String, region: String)
       extends BulkRemover {
-    val uri = new URI(storageNamespace)
-    val bucket = uri.getHost
+    private val uri = new URI(storageNamespace)
+    private val bucket = uri.getHost
 
     override def deleteObjects(keys: Seq[String], storageNamespace: String): Seq[String] = {
       val removeKeyNames = constructRemoveKeyNames(keys, storageNamespace, false, false)
-      println(s"Remove keys from ${bucket}: ${removeKeyNames.take(100).mkString(", ")}")
+      println(s"Remove keys from $bucket: ${removeKeyNames.take(100).mkString(", ")}")
       val removeKeys = removeKeyNames.map(k => new model.DeleteObjectsRequest.KeyVersion(k)).asJava
-
       val delObjReq = new model.DeleteObjectsRequest(bucket).withKeys(removeKeys)
       val s3Client = getS3Client(hc, bucket, region, S3NumRetries)
-      val res = s3Client.deleteObjects(delObjReq)
+      val deleteFun = generateDeleteObjectsFunc(s3Client)
+      val res = deleteFun.apply(delObjReq)
       res.getDeletedObjects.asScala.map(_.getKey())
     }
 
+    private def generateDeleteObjectsFunc(s3Client: AmazonS3): java.util.function.Function[DeleteObjectsRequest, DeleteObjectsResult] = {
+      val intervalFn =
+        IntervalFunction.ofExponentialRandomBackoff(500, 2, 0.5)
+      val retryConfig = RetryConfig.custom()
+        .maxAttempts(2)
```

Comment:
I don't understand: why specify an _exponential_ backoff if we're just going to try twice?  IIUC this configures a single interval - it would be easier for me to understand if we just specified it explicitly.

(Alternatively, trying more than twice also makes sense...)